[AOTI] add float support of triton

sijiac opened the pull request:
Summary: As the title

Test Plan: buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/dynamo:test_dynamo -- --exact 'caffe2/test/dynamo:test_dynamo - test_functions.py::DefaultsTests::test_triton_kernel_None_arg' --print-passing-details

Differential Revision: D51421325




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

oulgen commented on test/dynamo/test_functions.py:
```diff
@@ -1995,11 +1995,12 @@ def call_triton(x: torch.Tensor):
     @requires_cuda()
     @skipIfRocm
-    def test_triton_kernel_None_arg(self):
+    def test_triton_kernel_None_Float_arg(self):
```
test_triton_kernel_various_args

oulgen commented:
I landed my changes, please rebase past https://github.com/pytorch/pytorch/pull/114002

sijiac committed:
[AOTI] add float support of triton (#114014)

Summary:

As the title

Test Plan: buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/dynamo:test_dynamo -- --exact 'caffe2/test/dynamo:test_dynamo - test_functions.py::DefaultsTests::test_triton_kernel_None_arg' --print-passing-details

Reviewed By: oulgen

Differential Revision: D51421325

sijiac commented:
rebase to the master branch

sijiac commented:
@pytorchbot merge
